src/main/java/net/minecraftforge/items/wrapper/PlayerOffhandInvWrapper.java

```diff
@@ -0,0 +1,49 @@
+package net.minecraftforge.items.wrapper;
+
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.item.ItemStack;
+
+public class PlayerOffhandInvWrapper extends InvWrapper
+{
+    public final InventoryPlayer inventoryPlayer;
+    private final int offset;
+
```

Comment:
Interestiung, may be worth making a superclass that is InvWrapperRanged
and have a min/max check in it. As all 3 have somewhat similar restrictions.